In [1]:
# Calculate the probability of winning at least one block

invested_me = 1     # How many STX committed to each block
N_blocks = 100      # How many blocks mined
invested_others = 5000   # How many STX committed by others

P = 1 - (invested_others/(invested_others+invested_me))**N_blocks

print("Total investment:", N_blocks*invested_me, "STX.")
print("Probability of winning at least one block:",round(P*100,2),"%")

Total investment: 100 STX.
Probability of winning at least one block: 1.98 %


Unless you have a lot to commit, it is expensive to try to win blocks on your own, and random chance can make a very large difference in your actual return.  With enough funds and spreading out over a lot of blocks, you should get a fairly predictable amount of City Coins.  But without that, i.e., with small numbers, it's just gambling.  There are many other ways to gamble that are much simpler than using CityCoins.

So, if you don't have a lot to commit, it makes sense to join a mining pool.

In [2]:
# Estimate the return on winning and stacking CityCoins, in just one cycle.

# This calculates the return in terms of STX committed at the beginning.  The fluctuating price of STX 
# is an additional risk to this investment.

# There are many assumptions and caveats:
#
# This assumes perfect odds, where you get a fair share of total coins, proportional to what you pay in.
# Obviously if you win more or fewer coins than this, your return adjusts proportionally.
# Assume the average amount spent by everyone per block is constant (big, crude assumption!).
# It also assumes you immediately stack all of whatever you win into the subsequent cycle.
# Assume some constant fraction of all CityCoins will be stacked by everyone.

cycle_win = 0   # Cycle you earn your winnings (and thus stack in the next cycle)
N_cycles = 24   # Number of stacking cycles to compute, each lasting ~2 weeks.

frac_winnings_stacked = 0.8  # Fraction of total number of coins stacked by others


n_blockspercycle = 2100  # Number of blocks in each cycle (ignoring 100-block lag affecting only cycle 0)
tot_interest = 0
N_coins = 0

print("Interest is expressed as a fraction of the principle amount used to mine coins.")
print("E.g., you get your money back at 1.0, double your money at 2.0, etc.")

# This loop is set up only to work for the first ~4 years (after which rewards decrease further).

for i in range(1,N_cycles+1):   # Loop over cycles

    N_coins_incycle = 0
    for j in range(1,n_blockspercycle+1):   # Loop over blocks
        if (i-1)*n_blockspercycle + j < 10000:   # Bonus blocks
            coins_per_block = 250000 
        else:                                    # For next ~4 years
            coins_per_block = 100000
        N_coins_incycle += coins_per_block   # Total coins in cycle previous to stacking cycle
    N_coins += N_coins_incycle    # Total coins overall

#     print(N_coins_incycle, N_coins)

    N_coins_infirstcycle = n_blockspercycle * 250000    # Assumes rewards during bonus cycles only.
    new_interest = 0.7 * N_coins_infirstcycle / (frac_winnings_stacked*N_coins) 
    if i > cycle_win:
        tot_interest += new_interest
    print("cycle",i, " Your stacking cycle",i-cycle_win, \
          "  Interest =",round(new_interest,3), "  Total interest =",round(tot_interest,3))

Interest is expressed as a fraction of the principle amount used to mine coins.
E.g., you get your money back at 1.0, double your money at 2.0, etc.
cycle 1  Your stacking cycle 1   Interest = 0.875   Total interest = 0.875
cycle 2  Your stacking cycle 2   Interest = 0.438   Total interest = 1.312
cycle 3  Your stacking cycle 3   Interest = 0.292   Total interest = 1.604
cycle 4  Your stacking cycle 4   Interest = 0.219   Total interest = 1.823
cycle 5  Your stacking cycle 5   Interest = 0.18   Total interest = 2.003
cycle 6  Your stacking cycle 6   Interest = 0.166   Total interest = 2.17
cycle 7  Your stacking cycle 7   Interest = 0.155   Total interest = 2.324
cycle 8  Your stacking cycle 8   Interest = 0.144   Total interest = 2.469
cycle 9  Your stacking cycle 9   Interest = 0.136   Total interest = 2.604
cycle 10  Your stacking cycle 10   Interest = 0.128   Total interest = 2.732
cycle 11  Your stacking cycle 11   Interest = 0.121   Total interest = 2.852
cycle 12  Your stacking 

The above just calculates the return on stacking coins.  But you also still have the coins themselves, which might be worth nothing, but likely will trade for some value.

The implied mining cost of a coin is just the average amount put up by all for each block, divided by the reward.  The reward is 250k coins in the first 10,000 blocks (around 2-3 months), then it goes down to 100k for the next ~4 years.  It then halves every ~4 years for around 20 years or so, before flattening out at 3125 coins per block forever.

So, if we ASSUME that the amount people put up for each block is approximately constant (a big assumption), then the implied mining cost increases by a factor of 2.5 after the bonus round, then doubles every 4 years.  Doubling every 4 years is roughly like an annual, compounding growth rate of 20%.

For example, Miami Coin is tradeable on Okcoin, and its price so far (after just \~3 months) has approximately tracked the implied mining cost.  They typically have \~1000 STX (\~\\$2k) committed per block, and the price fluctuated a bunch in the first couple of weeks and is now \\$0.02 (which is the mining cost for 100k coins won per block).

NYC Coin in cycle 0 has been doin an average of approximately 5000-6000 STX per block.  This implies a mining cost of \~\\$0.04 now and going up to \~\\$0.10 in a couple of months, when the reward goes to 100k coins per block.

For example, if you put in 1000 STX (\~\\$2300), you might expect to earn \~42000 NYC Coins in the Syvita pool.  So maybe the sale price of those coins could be \~\\$4200 in a few months.  